frontend/app/src/pages/main/v1/workflow-runs-v1/$run/v2components/waterfall.tsx

```diff
@@ -0,0 +1,693 @@
+  const sortedRootTasks = rootTasks
+    .map((rootId) => ({ rootId, task: taskMap.get(rootId) }))
+    .filter(({ task }) => task !== undefined)
+    .sort((a, b) => (a.task!.taskId || 0) - (b.task!.taskId || 0))
+    .map(({ rootId }) => rootId);
+
+  sortedRootTasks.forEach((rootId) => {
+    addTasksPreorder(rootId);
+  });
+
+  return result;
+}
+
+// Status color configuration compatible with v1
+const StatusColors: Record<V1TaskStatus, string> = {
+  [V1TaskStatus.COMPLETED]: '#10b981', // green-500
+  [V1TaskStatus.FAILED]: '#ef4444', // red-500
+  [V1TaskStatus.CANCELLED]: '#ef4444', // red-500
+  [V1TaskStatus.RUNNING]: '#f59e0b', // amber-500
+  [V1TaskStatus.QUEUED]: '#6b7280', // gray-500
+};
+
+interface ProcessedTaskData {
+  id: string;
+  taskExternalId: string;
+  workflowRunId?: string;
+  taskDisplayName: string;
+  parentId?: string;
+  hasChildren: boolean;
+  depth: number;
+  isExpanded: boolean;
+  offset: number;
+  queuedDuration: number | null;
+  ranDuration: number | null;
+  status: V1TaskStatus;
+  taskId: number;
+  attempt: number;
+}
+
+interface ProcessedData {
+  data: ProcessedTaskData[];
+  taskPathMap: Map<string, string[]>;
+}
+
+interface WaterfallProps {
+  workflowRunId: string;
+  selectedTaskId?: string;
+  handleTaskSelect?: (taskId: string, childWorkflowRunId?: string) => void;
+}
+
+const CustomTooltip = (props: {
+  active?: boolean;
+  payload?: Array<{
+    dataKey: string;
+    name: string;
+    value: number;
+    payload?: V1TaskTiming;
+    [key: string]: unknown;
+  }>;
+  [key: string]: unknown;
+}) => {
+  const { active, payload } = props;
+
+  if (active && payload?.length) {
+    const filteredPayload = payload.filter(
+      (entry) => entry.dataKey !== 'offset' && entry.name !== 'Offset',
+    );
+
+    if (filteredPayload.length > 0) {
+      filteredPayload.forEach((entry, i) => {
+        if (filteredPayload[i].payload && entry.dataKey === 'ranDuration') {
+          const taskStatus = filteredPayload[0]?.payload?.status;
+
+          if (taskStatus && StatusColors[taskStatus]) {
+            filteredPayload[i].color = StatusColors[taskStatus];
+          }
+        }
+      });
+
+      const modifiedProps = { ...props, payload: filteredPayload };
+
+      return (
+        <ChartTooltipContent
+          {...modifiedProps}
+          className="w-[150px] sm:w-[200px] font-mono text-xs sm:text-xs cursor-pointer"
+        />
+      );
+    }
+  }
+
+  return null;
+};
+
+export function Waterfall({
+  workflowRunId,
+  selectedTaskId,
+  handleTaskSelect,
+}: WaterfallProps) {
+  const [expandedTasks, setExpandedTasks] = useState<Set<string>>(new Set());
+  const [autoExpandedInitially, setAutoExpandedInitially] = useState(false);
+  const [depth, setDepth] = useState(2);
+
+  // Use v1 style queries instead of _next hooks
+  const taskTimingsQuery = useQuery({
+    ...queries.v1WorkflowRuns.listTaskTimings(workflowRunId, depth),
+    refetchInterval: 1000,
```

Comment:
do we want to reduce this interval if complete?

Reply:
just reduced it to 5 seconds for now, I agree that every second is probably a bit high